src/kubernetes.cc

```diff
@@ -453,6 +453,75 @@ KubernetesReader::GetPodAndContainerMetadata(
   return std::move(result);
 }
 
+std::vector<json::value> KubernetesReader::GetServiceList(
+    const std::string& cluster_name, const std::string& location) const
+    throw(json::Exception) {
+  std::lock_guard<std::mutex> lock(service_mutex_);
+  std::vector<json::value> service_list;
+  for (const auto& metadata_it : service_to_metadata_) {
+    // service_key is a std::pair containing (namespace_name, service_name).
```

Comment:
This seems really redundant... You could have said something like: `// The namespace name is the first component of the service key.`, but even that basically replicates the code in line 464.
Maybe just `// A service key consists of a namespace name and a service name.`?

Reply:
done